File: auth-adapter-in/src/main/java/com/example/auth/adapter/in/rest/RegisterRequest.java
package com.example.auth.adapter.in.rest;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

public record RegisterRequest(
        @NotBlank String tenantSlug,
        @NotBlank @Email String email,
        @NotBlank @Size(min = 12, max = 128) String password,
        String deviceLabel) {
}

File: auth-adapter-in/src/main/kotlin/com/example/auth/adapter/in/rest/RegisterRequest.kt
package com.example.auth.adapter.`in`.rest

import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

@JvmRecord
data class RegisterRequest(
    @field:NotBlank val tenantSlug: String,
    @field:NotBlank @field:Email val email: String,
    @field:NotBlank @field:Size(min = 12, max = 128) val password: String,
    val deviceLabel: String?,
)
